pkg/httpserver/settings.go

```diff
@@ -0,0 +1,77 @@
+		// Exclude files by path, extension, or regular expression from being considered for compression.
+		// Useful if you are serving a format unknown to Gosoline.
+		Exclude CompressionExcludeSettings `cfg:"exclude"`
+	}
+
+	// CompressionExcludeSettings allow enabling of gzip support.
+	CompressionExcludeSettings struct {
+		Extension []string `cfg:"extension"`
+		Path      []string `cfg:"path"`
+		PathRegex []string `cfg:"path_regex"`
+	}
+
+	HealthCheckSettings struct {
+		Port    int             `cfg:"port"    default:"8090"`
+		Path    string          `cfg:"path"    default:"/health"`
+		Timeout TimeoutSettings `cfg:"timeout"`
+	}
+
+	LoggingSettings struct {
+		RequestBody       bool `cfg:"request_body"`
+		RequestBodyBase64 bool `cfg:"request_body_base64"`
+	}
+
+	ProfilingSettings struct {
+		Enabled bool                 `cfg:"enabled" default:"false"`
+		Api     ProfilingApiSettings `cfg:"api"`
+	}
+
+	ProfilingApiSettings struct {
+		Port int `cfg:"port" default:"8091"`
+	}
+	// Settings structure for an API server.
+	Settings struct {
+		// Port the API listens to.
+		Port string `cfg:"port"        default:"8080"`
+		// Mode is either debug, release, test.
+		Mode string `cfg:"mode"        default:"release" validate:"oneof=release debug test"`
+		// Compression settings.
+		Compression CompressionSettings `cfg:"compression"`
+		// Timeout settings.
+		Timeout TimeoutSettings `cfg:"timeout"`
+		// Logging settings
+		Logging LoggingSettings `cfg:"logging"`
+	}
+
+	// TimeoutSettings configures IO timeouts.
+	TimeoutSettings struct {
+		// You need to give at least 1s as timeout.
+		// Read timeout is the maximum duration for reading the entire request, including the body.
+		Read time.Duration `cfg:"read"     default:"60s" validate:"min=1000000000"`
+		// Write timeout is the maximum duration before timing out writes of the response.
+		Write time.Duration `cfg:"write"    default:"60s" validate:"min=1000000000"`
+		// Idle timeout is the maximum amount of time to wait for the next request when keep-alives are enabled
+		Idle time.Duration `cfg:"idle"     default:"60s" validate:"min=1000000000"`
+		// Drain timeout is the maximum amount of time to wait after receiving the kernel stop signal and actually shutting down the server
+		Drain time.Duration `cfg:"drain"    default:"0"   validate:"min=0"`
+		// Shutdown timeout is the maximum amount of time to wait for serving active requests before stopping the server
+		Shutdown time.Duration `cfg:"shutdown" default:"60s" validate:"min=1000000000"`
+	}
+
+	// TrafficDistributorSettings configures the traffic distributor middleware controlling maximum life of client connections.
+	TrafficDistributorSettings struct {
+		Enabled                   bool          `cfg:"enabled"                   default:"false"`
+		MaxConnectionAge          time.Duration `cfg:"maxConnectionAge"          default:"1m"`
+		MaxConnectionRequestCount int           `cfg:"maxConnectionRequestCount" default:"0"`
+	}
```

Comment:
I don't see them in our settings struct, which seems wrong. I am also not sure about the defaults - why isn't it enabled by default? I would also give them a different name, maybe ConnectionLimitSettings? Next, please use snake_case for setting names. Finally, I would shorten the names to maximum_age or max_age and maximum_requests or max_requests, as connection_limits already establish the context of a connection